apps/passwd.c

```diff
@@ -260,7 +265,7 @@ int passwd_main(int argc, char **argv)
         do {                    /* loop over list of passwords */
             passwd = *passwds++;
             if (!do_passwd(passed_salt, &salt, &salt_malloc, passwd, bio_out,
-                           quiet, table, reverse, pw_maxlen, mode))
+                           quiet, table, reverse, pw_maxlen, mode, rounds))
```

Comment:
Please add a check before the loop that rounds is used only with SHA-based algorithms. Output an error message and return failure otherwise.